src/material/datepicker/calendar-body.scss

```diff
@@ -31,7 +31,12 @@ $calendar-range-end-body-cell-size:
   padding-right: $calendar-body-label-side-padding;
 }
 
+.mat-calendar-body-cell-wrapper {
+  display: contents;
+}
+
 .mat-calendar-body-cell {
+  display: table-cell;
```

Comment:
We had an issue where NVDA would announce an extra "table" because something had `display: table`. I wouldn't be surprised if it becomes an issue here too. See https://github.com/angular/components/pull/23446.